### PR TITLE
Rename BL_Python to Ligare

### PR DESCRIPTION
**Pull request by aholmes**
Per https://github.com/uclahs-cds/group-CDS-owners/issues/111, BL_Python will be renamed to Ligare. This PR contains replacement values for strings matching (case-insensitive) `BL_Python`, `BL-Python`, `BL_Alembic`, `BL-Alembic`, `BLAlembic`. These replacements include file names, directory names, variables, imports, strings, modules, configurations, etc. There should be no remaining references to `BL.*` in this PR (this regex isn't accurate, but close enough).

### CODE REVIEW

File: src/programming/test/unit/patterns/test_dependency_injection.py [hunk 1]
```diff
@@ -2,12 +2,9 @@
...
 from logging import Logger
 
-from BL_Python.programming.patterns.dependency_injection import (
-    BatchModule,
-    LoggerModule,
-)
-from BL_Python.programming.str import get_random_str
 from injector import Injector
+from Ligare.programming.patterns.dependency_injection import BatchModule, LoggerModule
+from Ligare.programming.str import get_random_str
```
Comment (nwiltsie) [inline src/programming/test/unit/patterns/test_dependency_injection.py:7]: Really, ruff? The line becomes 3 characters shorter and it therefore has to be reflowed?

File: src/database/Ligare/database/migrations/alembic/__main__.py [hunk 2]
```diff
@@ -2,30 +2,30 @@
...
 
-def bl_alembic(
+def ligare_alembic(
     argv: list[str] | None = None,
     log_level: int | str | None = None,
     allow_overwrite: bool | None = None,
 ) -> None:
     """
-    A method to support the `bl-alembic` command, which replaces `alembic.
+    A method to support the `ligare-alembic` command, which replaces `alembic`.
```
Comment (nwiltsie) [inline src/database/Ligare/database/migrations/alembic/__main__.py:14]: This entire PR was worth it to catch this typo.

**Review by nwiltsie — APPROVED** [review]
Whew!

Big shout-out to [diffr](https://github.com/mookid/diffr), it highlights word-level differences and made this _so_ much easier to skim through with `gh pr diff 115 | diffr`.

Screenshot 2024-09-24 at 12 53 34 PM

File: src/programming/test/unit/patterns/test_dependency_injection.py (hunk 1 above)
Comment (nwiltsie) [reply src/programming/test/unit/patterns/test_dependency_injection.py:7]: Snark aside, it was hugely helpful to do the reformatting in a different commit - awesome job.

Comment (aholmes) [reply src/programming/test/unit/patterns/test_dependency_injection.py:7]: That's how it do be when only the tools have opinions.

File: src/database/Ligare/database/migrations/alembic/__main__.py (hunk 2 above)
Comment (aholmes) [reply src/database/Ligare/database/migrations/alembic/__main__.py:14]: I'll open a second PR for just the typo. ;)

**Pull request merged**